### PR TITLE
W-11365512: Fix parsing of repeated array headers with quotes

### PR DESCRIPTION
**Pull request by dcourtil**
This is a continuation and fix for the initial change done as part of https://github.com/mulesoft/apikit-rest-module/pull/590

It restores the tests cases assertions that were changed by the previous commit (see https://github.com/mulesoft/apikit-rest-module/pull/590#discussion_r913873466) and it updates the parser so the quotes handling is consistent between both strategies.

It is worth to note that additional changes are introduce by this fix:
- Quotes between quotes need to be escaped.
- If unbalanced quotes are found, the default parser returns the value as it is
- As like HTTP listener does, a blank or empty string value is passed as a simple empty string unless it is between quotes.
- Fixes an issue on `HeaderValidator` which was failing to remove the original list of header values due a mismatch on the header name case. Therefore, the array header values were passed to the flow duplicated (with same name but different cases).

### CODE REVIEW

File: src/main/java/org/mule/module/apikit/deserializing/ArrayHeaderAttributeDeserializer.java [hunk 1]
```diff
@@ -32,33 +24,38 @@ public ArrayHeaderAttributeDeserializer(ArrayHeaderDelimiter arrayHeaderDelimite
...
           curVal.append(ch);
+          escapedChar = false;
         } else if (ch == CLOSING_CURLY_BRACE) {
           curlyBracesStackCount--;
           curVal.append(ch);
+          escapedChar = false;
         } else {
           curVal.append(ch);
+          escapedChar = false;
         }
```
Comment (iglosiggio) [inline src/main/java/org/mule/module/apikit/deserializing/ArrayHeaderAttributeDeserializer.java:59]: I think that strings like `asd"{{{"asd}}}` will not work as expected. It also looks a bit weird that `"\{\{"` is treated exactly as `"{{"`

Also, maybe cleaner (don't know):
```suggestion
        if (ch == '\\') {
          escapedChar = true;
        } else if (ch == DOUBLE_QUOTES && !escapedChar) {
          inQuotes = false;
        } else {
          if (ch == OPENING_CURLY_BRACE) {
            curlyBracesStackCount++;
          } else if (ch == CLOSING_CURLY_BRACE) {
            curlyBracesStackCount--;
          }
          escapedChar = false;
          curVal.append(ch);
        }
```

**Comment by iglosiggio** [comment]
Can't comment this inline because it refers to older code:

ArrayHeaderAttributeDeserializer.java:60
```java
      } else if (curlyBracesStackCount > 0) {
        if (ch == CLOSING_CURLY_BRACE) {
          curlyBracesStackCount--;
        } else if (ch == OPENING_CURLY_BRACE) {
          curlyBracesStackCount++;
        }
        if (curlyBracesStackCount >= 0) {
          curVal.append(ch);
        }
      } else {
```

`curlyBracesStackCount >= 0` is an unnecessary check and should be removed. If `curlyBracesStackCount > 0` at the start then after we decreasing or increasing it by one `curlyBracesStackCount >= 0` will always be true 😛.

**Comment by dcourtil** [comment]
> * If unbalanced quotes are found, the default parser returns the value as it is

True! 😅

**Comment by dcourtil** [comment]
> Can't comment this inline because it refers to older code:
> 
> ArrayHeaderAttributeDeserializer.java:60
> 
> ```java
>       } else if (curlyBracesStackCount > 0) {
>         if (ch == CLOSING_CURLY_BRACE) {
>           curlyBracesStackCount--;
>         } else if (ch == OPENING_CURLY_BRACE) {
>           curlyBracesStackCount++;
>         }
>         if (curlyBracesStackCount >= 0) {
>           curVal.append(ch);
>         }
>       } else {
> ```
> 
> `curlyBracesStackCount >= 0` is an unnecessary check and should be removed. If `curlyBracesStackCount > 0` at the start then after we decreasing or increasing it by one `curlyBracesStackCount >= 0` will always be true 😛.

True! 😅

**Comment by iglosiggio** [comment]
Alternative implementation for the main loop of `ArrayHeaderAttributeDeserializer#deserializeValue(String)`:
```java
for (char ch : chars) {
  if (insideQuotes) {
    if (ch == '\\') {
      escapedChar = true;
    } else if (ch == DOUBLE_QUOTES && !escapedChar) {
      insideQuotes = false;
    } else {
      curVal.append(ch);
      escapedChar = false;
    }
  } else if (/* inside braces */ curlyBracesStackCount > 0) {
    if (ch == CLOSING_CURLY_BRACE) {
      curlyBracesStackCount--;
    } else if (ch == OPENING_CURLY_BRACE) {
      curlyBracesStackCount++;
    }
    curVal.append(ch);
  } else {
    if (ch == DOUBLE_QUOTES) {
      insideQuotes = true;
    } else if (ch == OPENING_CURLY_BRACE) {
      curlyBracesStackCount++;
      curVal.append(ch);
    } else if (ch == delimiter) {
      addValueToList(values, curVal);
      curVal = new StringBuffer();
    } else if (ch != '\r' && ch != '\n') {
      curVal.append(ch);
    }
  }
}
```

File: src/main/java/org/mule/module/apikit/deserializing/ArrayHeaderAttributeDeserializer.java (hunk 1 above)
Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/deserializing/ArrayHeaderAttributeDeserializer.java]: I updated the logic and added some tests around the cases that you mentioned

**Comment by dcourtil** [comment]
@iglosiggio @JuanAller I updated the logic of the deserializers, keeping the behavior that I mentioned. Hope this is more clear now.

**Review by iglosiggio — APPROVED** [review]

**Pull request merged**